### ISSUE TITLE
Java 8 - Could not instantiate mapper : com.thoughtworks.xstream.mapper.LambdaMapper

### ISSUE THREAD

**Issue by steve-white** [x-stream/xstream#30]
Hi, we're having an issue with xstream 1.4.8 (and 1.5.0-SNAPSHOT) on Java 8 but not Java 7. We're literally just calling the constructor which appears to loading lambda handler classes dynamically from the class loader when Java 8 is used. 

Is this a known issue or should we be handling this differently?

The code to trigger this is a simple console app with the following in main:
```XStream xs = new XStream();```

Trace:
```com.thoughtworks.xstream.InitializationException: Could not instantiate mapper : com.thoughtworks.xstream.mapper.LambdaMapper : com.thoughtworks.xstream.mapper.LambdaMapper
        at com.thoughtworks.xstream.XStream.buildMapperDynamically(XStream.java:595)
        at com.thoughtworks.xstream.XStream.buildMapper(XStream.java:575)
        at com.thoughtworks.xstream.XStream.<init>(XStream.java:544)```

**Comment by joehni** [comment]
[Travis](https://travis-ci.org/x-stream/xstream/branches) demonstrates, that this is no issue. So what's different with your environment?

**Comment by steve-white** [comment]
Right I've found the issue. After some more testing our simple test app works fine using the v1.4.8 artifact from maven central but the issue occurs when we've built xstream from source. We build using JDK 7, historically to keep backward compatibility with certain libs, which is the problem.
 
Building xstream against JDK 7 doesn't build the XStream Lambda classes (which is fine because we don't use them), but when that artifact is run on a Java 8 runtime it assumes they're available and so tries to load them. I've just set our CI to build it using JDK 8.
